testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -1978,6 +1978,38 @@ public void testCreateTable()
         assertFalse(getQueryRunner().tableExists(getSession(), tableNameLike));
     }
 
+    @Test
+    public void testCreateTableWithLongTableName()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
+
+        String baseTableName = "test_create_" + randomTableSuffix();
+
+        if (maxTableNameLength().isEmpty()) {
+            // assume 65535 is enough for most use cases
+            String tableName = baseTableName + "z".repeat(65535);
```

Comment:
This creates name longer than 65535. Intentional? Comment suggests it isn't.

the `else` branch takes care to create name of precisely given length, so i would unify them with
```
int maxLength = maxTableNameLength()
        // Assume 2^16 is enough for most use cases. Add a bit more to ensure 2^16 isn't actual limit.
        .orElse(65536 + 5);
```